server/src/main/java/org/opensearch/action/search/SearchExecutionStatsCollector.java

```diff
@@ -67,6 +67,19 @@
     @Override
     public void onResponse(SearchPhaseResult response) {
         QuerySearchResult queryResult = response.queryResult();
+        if (response.getShardSearchRequest() != null) {
+            if (response.remoteAddress() != null) {
+                // update outbound network time for request sent over network for shard requests
+                response.getShardSearchRequest()
+                    .setOutboundNetworkTime(
+                        Math.max(0, System.currentTimeMillis() - response.getShardSearchRequest().getOutboundNetworkTime())
+                    );
+            } else {
```

Comment:
To measure time elapsed you should always use a monotonic logical clock time. Uses `nanoTime` instead
https://docs.oracle.com/javase/7/docs/api/java/lang/System.html#nanoTime()

Reply:
For time elapsed on a single machine, nanoTime is the best choice as it returns accurate results. However, since this is across different machines, nanoTime returns nanoseconds since some fixed but arbitrary time (in the past/future) that is different for different vms. 
That's why for meaningful results for network time in multi-node clusters, I've used System.currentTimeMillis() instead which is more accurate.